### ISSUE TITLE
Infinite loop when parsing invalid HTML

### ISSUE THREAD

**Issue by PWalkow** [paquettg/php-html-parser#187]
I found a piece of invalid html that is causing infinite loop (and maximum execution time issues). 
similiar issue was #179 that was closed already but this example shows that similiar cases exist, how we could fix it without having access to change the html?

```
<img src="www.example.com/image.jpg" data-caption="How did <i>
```

I understand it is an invalid html but the parser should not stuck when trying to parse it and fail gracefully rather than causing infinite execution time

**Comment by paquettg** [comment]
Thanks for the report.

I have added this to the release in 2.1.1 and, once I confirm the issue with a test, I'll see to a fix.

If you can provide a PR with a fix and test I would also appreciate that.

Thank you.

**Comment by EnriqueGF** [comment]
I'm having the same issue with a broken html.

**Comment by zrlhk** [comment]
I update to 2.1.1 and it's still Infinite loop

**Comment by Gicminos** [comment]
I think I have the same problem with this url: https://www.business-solutions.telefonica.com/en/products/security/cloud-security/

**Comment by tfedor** [comment]
@paquettg The issue is caused by parsing attribute where you don't check length of file; here: https://github.com/paquettg/php-html-parser/blob/master/src/PHPHtmlParser/Dom.php#L760 and here: https://github.com/paquettg/php-html-parser/blob/master/src/PHPHtmlParser/Dom.php#L772

Fix should be easy, something like:
```
} while (!empty($moreString) && $this->content->getPosition() < $this->content->getSize());
```

I'd push a PR, but adding this fix caused this strspn to return false and I'm not sure what's that supposed to do:
```
https://github.com/paquettg/php-html-parser/blob/master/src/PHPHtmlParser/Content.php#L237
```

I'd just fix that by throwing LogicalException, but gonna leave that up to you.

**Comment by PWalkow** [comment]
@paquettg 
I understand that some of the things were fixed to stop infinite loops but an example that I had in my integration tests with this library is still failing for:
```'<the thing broke itV'```

**Comment by paquettg** [comment]
I'll try this out on the 3.0.0 dev branch later tonight

**Comment by paquettg** [comment]
Fixed this issue in the 3.0.0 branch. Now the string `<the thing broke itv` will parse into `<the thing broke itv></the>` and will not result in an infinite loop.
